File: src/main/java/com/codecool/cashtrack/controllers/utils/ResponseUtil.java
package com.codecool.cashtrack.controllers.utils;

import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

public class ResponseUtil {

    public static <T> ResponseEntity<T> wrapEntity(T entity) {
        HttpHeaders headers = new HttpHeaders();
        headers.add("Content-Type", "application/json; charset=utf-8");
        return new ResponseEntity<>(entity, headers, HttpStatus.OK);
    }

    public static <T> ResponseEntity<T> wrapEntity(T entity, HttpStatus status) {
        HttpHeaders headers = new HttpHeaders();
        headers.add("Content-Type", "application/json; charset=utf-8");
        return new ResponseEntity<>(entity, headers, status);
    }
}

File: src/main/java/com/codecool/repvault/controllers/utils/ResponseUtil.kt
package com.codecool.repvault.controllers.utils

import org.springframework.http.HttpHeaders
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity

object ResponseUtil {
    fun <T : Any> wrapEntity(entity: T?): ResponseEntity<T> {
        val headers = HttpHeaders()
        headers.add("Content-Type", "application/json; charset=utf-8")
        return ResponseEntity<T>(entity, headers, HttpStatus.OK)
    }

    fun <T : Any> wrapEntity(entity: T?, status: HttpStatus): ResponseEntity<T> {
        val headers = HttpHeaders()
        headers.add("Content-Type", "application/json; charset=utf-8")
        return ResponseEntity<T>(entity, headers, status)
    }
}
